generator/src/main/java/com/scottlogic/deg/generator/constraints/StringConstraintsCollection.java

```diff
@@ -169,14 +200,62 @@ private AtomicConstraint getComplementOfConstraint(AtomicConstraint negatedConst
         if (negatedConstraint instanceof StringHasLengthConstraint){
             StringHasLengthConstraint ofLength = (StringHasLengthConstraint) negatedConstraint;
 
-            //can use either longerThan <length> or shorterThan <length>
-            return new IsStringShorterThanConstraint(
-                ofLength.field,
-                ofLength.referenceValue,
-                ofLength.getRules()
-            );
+            return new StringHasDifferentLengthConstraint(ofLength);
         }
 
         return null;
     }
+
+    private class StringHasDifferentLengthConstraint implements AtomicConstraint {
```

Comment:
Should we just make this a regular atomicConstraint. rather than a private one?

Reply:
Its only an atomic constraint so it can fit with the others. It is exactly the same as `not(ofLength XX)`. Its just in this implementation we cannot deal with 'notted' constraints, hence the explicit version.

I think adding this to the schema would add confusion, and extra complexity with little benefit. i.e. `not(notOfLength XXX)` === `ofLength XXX` and so-on